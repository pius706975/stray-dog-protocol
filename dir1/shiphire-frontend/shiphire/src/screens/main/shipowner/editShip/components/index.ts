export {default as SpecificFormInputField} from './SpecificFormInputField';
export {default as SizeForm} from './SizeForm';
export {default as EditCustomDropDownInput} from './EditCustomDropDownInput';
