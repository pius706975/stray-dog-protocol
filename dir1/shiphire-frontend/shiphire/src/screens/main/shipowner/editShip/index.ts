export {default as EditGeneralForm} from './EditGeneralForm';
export {default as EditSpecificForm} from './EditSpecificForm';
export {default as EditImageForm} from './EditImageForm';
export {default as EditDocumentForm} from './EditDocumentForm';
