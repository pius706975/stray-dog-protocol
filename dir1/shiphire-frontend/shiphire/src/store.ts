import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
