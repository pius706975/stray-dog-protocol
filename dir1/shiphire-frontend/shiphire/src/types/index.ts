export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'