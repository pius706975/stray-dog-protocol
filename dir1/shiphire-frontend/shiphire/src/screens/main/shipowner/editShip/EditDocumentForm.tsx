import mime from 'mime';
import React from 'react';
import DocumentPicker, {
    DocumentPickerResponse,
} from 'react-native-document-picker';
import { Text, TouchableOpacity, View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    ArrowIcon,
    CloseIcon,
    Color,
    FontFamily,
    FontSize,
    PlusIcon,
} from '../../../../configs';
import { useEditShipDocument } from '../../../../hooks';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { EditDocumentFormProps } from '../../../../types';
import { useTranslation } from 'react-i18next';

const EditDocumentForm: React.FC<EditDocumentFormProps> = ({
    navigation,
    route,
}) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('detailship');
    const { shipId, documentName, documentUrl } = route.params;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const { showModal, hideModal } = modalSlice.actions;
    const [selectedPdf, setSelectedPdf] =
        React.useState<DocumentPickerResponse | null>(null);
    const [showDeleteButton, setShowDeleteButton] = React.useState(false);
    const mutationEditShipDocument = useEditShipDocument();
    const documentFormData = new FormData();

    const openPdfPickerBusiness = () => {
        DocumentPicker.pick({
            type: [DocumentPicker.types.pdf],
        })
            .then(result => {
                setSelectedPdf(result[0]);
                setShowDeleteButton(true);
            })
            .catch(error => {
                console.log('Error selecting PDF:', error);
            });
    };

    const handleSubmit = () => {
        dispatch(showProgressIndicator());
        if (selectedPdf) {
            if (selectedPdf.name == documentName) {
                dispatch(hideProgressIndicator());
                dispatch(
                    showModal({
                        status: 'failed',
                        text: t('ShipOwner.EditDocumentForm.failedPdfNameSame'),
                    }),
                );
                setTimeout(() => {
                    dispatch(hideModal());
                }, 2000);
            } else {
                documentFormData.append('files', {
                    uri: selectedPdf.uri,
                    type: mime.getType(selectedPdf.name),
                    name: `Ship ${selectedPdf.name} Owner Document 1`,
                });
                mutationEditShipDocument.mutate(
                    {
                        request: documentFormData,
                        shipId,
                    },
                    {
                        onSuccess: resp => {
                            dispatch(hideProgressIndicator());
                            dispatch(
                                showModal({
                                    status: 'success',
                                    text: t(
                                        'ShipOwner.EditDocumentForm.successEditDocument',
                                    ),
                                }),
                            );
                            navigation.navigate('OwnerDetailShip', {
                                shipId: resp.data.data.shipId,
                            });
                            setTimeout(() => {
                                dispatch(hideModal());
                            }, 2000);
                        },
                    },
                );
            }
        } else {
            dispatch(hideProgressIndicator());
            dispatch(
                showModal({
                    status: 'failed',
                    text: t('ShipOwner.EditDocumentForm.failedPdfNameSame'),
                }),
            );
            setTimeout(() => {
                dispatch(hideModal());
            }, 2000);
        }
    };

    React.useEffect(() => {
        const initialShipDocumentData = {
            uri: 'string',
            name: documentName,
            copyError: '',
            fileCopyUri: '',
            type: 'application/pdf',
            size: 0,
        };
        if (documentName !== null) {
            setSelectedPdf(initialShipDocumentData);
            setShowDeleteButton(true);
        }
    }, []);

    return (
        <ScreenLayout backgroundColor={'light'} testId={'documentFormAddShip'}>
            <View marginH-20 marginV-26>
                <View>
                    <View
                        style={{
                            paddingBottom: 5,
                        }}>
                        <Text
                            style={{
                                fontFamily: FontFamily.medium,
                                fontSize: FontSize.xl,
                                color: Color.primaryColor,
                            }}>
                            {t('ShipOwner.EditDocumentForm.labelDocument1')}
                        </Text>
                    </View>
                    <View
                        style={{
                            paddingBottom: 25,
                        }}>
                        {selectedPdf ? (
                            <View
                                style={{
                                    flexDirection: 'row',
                                    alignItems: 'center',
                                    borderBottomWidth: 1,
                                    borderBottomColor: Color.primaryColor,
                                    paddingBottom: 15,
                                }}>
                                <Text style={{ flex: 1 }}>
                                    {selectedPdf.name}
                                </Text>
                                {showDeleteButton && (
                                    <TouchableOpacity
                                        onPress={() => {
                                            setSelectedPdf(null);
                                            setShowDeleteButton(false);
                                        }}>
                                        <CloseIcon />
                                    </TouchableOpacity>
                                )}
                            </View>
                        ) : (
                            <TouchableOpacity onPress={openPdfPickerBusiness}>
                                <View
                                    style={{
                                        flexDirection: 'row',
                                        alignItems: 'center',
                                        borderBottomWidth: 1,
                                        borderBottomColor: Color.primaryColor,
                                        paddingBottom: 15,
                                    }}>
                                    <Text style={{ flex: 1 }}>
                                        {' '}
                                        {t(
                                            'ShipOwner.EditImageForm.textSelectFile',
                                        )}
                                    </Text>
                                    <PlusIcon />
                                </View>
                            </TouchableOpacity>
                        )}
                    </View>

                    <View
                        style={{
                            flex: 1,
                            justifyContent: 'flex-end',
                        }}></View>
                </View>

                <Button
                    rightIcon={<ArrowIcon />}
                    title={t('ShipOwner.EditImageForm.labelButtonNext')}
                    onSubmit={() => handleSubmit()}
                />
            </View>
        </ScreenLayout>
    );
};

export default EditDocumentForm;
