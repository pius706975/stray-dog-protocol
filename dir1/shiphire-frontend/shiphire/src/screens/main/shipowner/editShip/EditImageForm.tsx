import mime from 'mime';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Text, TouchableOpacity } from 'react-native';
import ImagePicker, { ImageOrVideo } from 'react-native-image-crop-picker';
import { View } from 'react-native-ui-lib';
import { useDispatch, useSelector } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    ArrowIcon,
    CloseIcon,
    Color,
    FontFamily,
    FontSize,
    PlusIcon,
} from '../../../../configs';
import { useEditShipImage } from '../../../../hooks';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { EditImageFormProps, RootState } from '../../../../types';

const EditImageForm: React.FC<EditImageFormProps> = ({ navigation, route }) => {
    // const shipData = useSelector((state: RootState) => state.addShip);
    const { shipId, shipName } = route.params;
    const { t } = useTranslation('detailship');
    const [selectedImage, setSelectedImage] = React.useState<ImageOrVideo>();
    const [showDeleteBusinessButton, setShowDeleteBusinessButton] =
        React.useState(false);
    const mutationEditShipImage = useEditShipImage();
    const dispatch = useDispatch();
    const { showModal, hideModal } = modalSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const request = new FormData();

    const openImage = () => {
        ImagePicker.openPicker({
            width: 300,
            height: 400,
            cropping: true,
        })
            .then(image => {
                setSelectedImage(image);
            })
            .catch(e => console.log(e));
    };

    const handleSubmit = () => {
        dispatch(showProgressIndicator());

        if (selectedImage?.path === undefined) {
            dispatch(hideProgressIndicator());
            dispatch(
                showModal({
                    status: 'failed',
                    text: t('ShipOwner.EditImageForm.labelMustSelect'),
                }),
            );
            setTimeout(() => {
                dispatch(hideModal());
            }, 2000);
        } else if (selectedImage) {
            console.log(selectedImage?.path);

            request.append('image', {
                uri: selectedImage?.path,
                type: mime.getType(selectedImage?.path),
                name: `Image Ship ${shipName}`,
            });
            mutationEditShipImage.mutate(
                { request, shipId },
                {
                    onSuccess: () => {
                        dispatch(hideProgressIndicator());
                        dispatch(
                            showModal({
                                status: 'success',
                                text: t(
                                    'ShipOwner.EditImageForm.successShipImageEdited',
                                ),
                            }),
                        );
                        setTimeout(() => {
                            dispatch(hideModal());
                        }, 2000);

                        navigation.navigate('OwnerDetailShip', { shipId });
                    },
                    onError: () => {
                        dispatch(hideProgressIndicator());
                        dispatch(
                            showModal({
                                status: 'failed',
                                text: t(
                                    'ShipOwner.EditImageForm.failedEditShipImage',
                                ),
                            }),
                        );
                        setTimeout(() => {
                            dispatch(hideModal());
                        }, 2000);
                    },
                },
            );
        }
    };

    return (
        <ScreenLayout backgroundColor={'light'} testId={'imageFormAddShip'}>
            <View marginH-20 marginV-26>
                <View>
                    <View
                        style={{
                            paddingBottom: 5,
                        }}>
                        <Text
                            style={{
                                fontFamily: FontFamily.medium,
                                fontSize: FontSize.xl,
                                color: Color.primaryColor,
                            }}>
                            {t('ShipOwner.EditImageForm.textChooseImage')}
                        </Text>
                    </View>
                    <View
                        style={{
                            paddingBottom: 25,
                        }}>
                        {selectedImage ? (
                            <View
                                testID="selectedImage"
                                style={{
                                    flexDirection: 'row',
                                    alignItems: 'center',
                                    borderBottomWidth: 1,
                                    borderBottomColor: Color.primaryColor,
                                    paddingBottom: 15,
                                }}>
                                <Text
                                    style={{
                                        flex: 1,
                                        color: Color.darkTextColor,
                                    }}>
                                    {selectedImage.path.split('/').pop()}
                                </Text>
                                {showDeleteBusinessButton && (
                                    <TouchableOpacity
                                        onPress={() => {
                                            setSelectedImage(undefined);
                                            setShowDeleteBusinessButton(false);
                                        }}>
                                        <CloseIcon />
                                    </TouchableOpacity>
                                )}
                            </View>
                        ) : (
                            <TouchableOpacity
                                testID="selectImageButton"
                                onPress={openImage}>
                                <View
                                    style={{
                                        flexDirection: 'row',
                                        alignItems: 'center',
                                        borderBottomWidth: 1,
                                        borderBottomColor: Color.primaryColor,
                                        paddingBottom: 15,
                                    }}>
                                    <Text style={{ flex: 1 }}>
                                        {t(
                                            'ShipOwner.EditImageForm.textSelectFile',
                                        )}
                                    </Text>
                                    <PlusIcon />
                                </View>
                            </TouchableOpacity>
                        )}
                    </View>
                </View>
                <View
                    style={{
                        flex: 1,
                        justifyContent: 'flex-end',
                    }}>
                    <Button
                        testID="submitButton"
                        rightIcon={<ArrowIcon />}
                        title={t('ShipOwner.EditImageForm.labelButtonNext')}
                        onSubmit={() => handleSubmit()}
                    />
                </View>
            </View>
        </ScreenLayout>
    );
};

export default EditImageForm;
