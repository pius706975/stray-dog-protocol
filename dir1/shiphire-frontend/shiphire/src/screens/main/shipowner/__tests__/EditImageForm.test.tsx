import { NavigationContainer } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import httpRequest from '../../../../services/api';
import store from '../../../../store';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MainOwnerStackParamList } from '../../../../types';
import { EditImageForm } from '../editShip';
import {
    setMockImagePickerPathResolve,
    setMockImagePickerResolve,
} from '../../../../jest/setup';

const queryClient = new QueryClient();
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockParams = {
    shipId: '65a8bda44b191011546c21b1',
    shipName: 'Barge Hauler',
};

const editImageFormComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="EditImageForm"
                        component={EditImageForm}
                        initialParams={mockParams}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing Edit Image Form Screen', () => {
    describe('Snapshot Testing', () => {
        it('should render Edit Image Form correctly', () => {
            const editImageForm = render(editImageFormComponent);
            expect(editImageForm).toMatchSnapshot();
        });
    });

    describe('Unit Testing', () => {
        it('should handle success edit image ship', async () => {
            const { getByTestId } = render(editImageFormComponent);

            const selectImageButton = getByTestId('selectImageButton');
            const submitButton = getByTestId('submitButton');

            fireEvent.press(selectImageButton);
            setMockImagePickerPathResolve(true);

            fireEvent.press(submitButton);

            mockAdapter
                .onPost(`/ship-owner/edit-ship-image/${mockParams.shipId}`)
                .reply(200, {
                    message: 'Success',
                });

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'ShipOwner.EditImageForm.successShipImageEdited',
                    visible: true,
                });

                jest.advanceTimersByTime(2000);
            });
        });

        it('should handle error edit image ship', async () => {
            const { getByTestId } = render(editImageFormComponent);

            const selectImageButton = getByTestId('selectImageButton');
            const submitButton = getByTestId('submitButton');

            fireEvent.press(selectImageButton);
            setMockImagePickerPathResolve(true);

            fireEvent.press(submitButton);

            mockAdapter
                .onPost(`/ship-owner/edit-ship-image/${mockParams.shipId}`)
                .reply(500, {
                    message: 'Failed',
                });

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'failed',
                    text: 'ShipOwner.EditImageForm.failedEditShipImage',
                    visible: true,
                });

                jest.advanceTimersByTime(2000);
            });
        });
    });
});
