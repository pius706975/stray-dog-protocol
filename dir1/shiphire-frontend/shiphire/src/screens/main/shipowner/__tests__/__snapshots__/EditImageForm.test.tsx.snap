// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Edit Image Form Screen Snapshot Testing should render Edit Image Form correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="imageFormAddShip"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  marginH-20={true}
                  marginV-26={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {},
                      {
                        "marginHorizontal": 20,
                        "marginVertical": 26,
                      },
                      {},
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "paddingBottom": 5,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#216dab",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 22,
                          }
                        }
                      >
                        ShipOwner.EditImageForm.textChooseImage
                      </Text>
                    </View>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "paddingBottom": 25,
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                        testID="selectImageButton"
                      >
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "alignItems": "center",
                                "borderBottomColor": "#216dab",
                                "borderBottomWidth": 1,
                                "flexDirection": "row",
                                "paddingBottom": 15,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              {
                                "flex": 1,
                              }
                            }
                          >
                            ShipOwner.EditImageForm.textSelectFile
                          </Text>
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight="24"
                            bbWidth="24"
                            fill="none"
                            focusable={false}
                            height={24}
                            meetOrSlice={0}
                            minX={0}
                            minY={0}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 24,
                                  "width": 24,
                                },
                              ]
                            }
                            vbHeight={24}
                            vbWidth={24}
                            width={24}
                            xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_28470)">
    <path d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z" fill="#323232"/>
    </g>
    <defs>
    <clipPath id="clip0_77_28470">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
    
    
    "
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <RNSVGGroup
                              fill={null}
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            >
                              <RNSVGGroup
                                clipPath="clip0_77_28470"
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                              >
                                <RNSVGPath
                                  d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z"
                                  fill={
                                    {
                                      "payload": 4281479730,
                                      "type": 0,
                                    }
                                  }
                                  propList={
                                    [
                                      "fill",
                                    ]
                                  }
                                />
                              </RNSVGGroup>
                              <RNSVGDefs>
                                <RNSVGClipPath
                                  fill={
                                    {
                                      "payload": 4278190080,
                                      "type": 0,
                                    }
                                  }
                                  name="clip0_77_28470"
                                >
                                  <RNSVGRect
                                    fill={
                                      {
                                        "payload": 4294967295,
                                        "type": 0,
                                      }
                                    }
                                    height="24"
                                    propList={
                                      [
                                        "fill",
                                      ]
                                    }
                                    width="24"
                                    x="0"
                                    y="0"
                                  />
                                </RNSVGClipPath>
                              </RNSVGDefs>
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                          "justifyContent": "flex-end",
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#216dab",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 0,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="submitButton"
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        ShipOwner.EditImageForm.labelButtonNext
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      >
                        <RNSVGSvgView
                          align="xMidYMid"
                          bbHeight="20"
                          bbWidth="20"
                          fill="none"
                          focusable={false}
                          height={20}
                          meetOrSlice={0}
                          minX={0}
                          minY={0}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 20,
                                "width": 20,
                              },
                            ]
                          }
                          vbHeight={20}
                          vbWidth={20}
                          width={20}
                          xml="<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_382_280)">
    <path d="M4.16666 10.8334H13.475L9.40833 14.9001C9.08333 15.2251 9.08333 15.7584 9.40833 16.0834C9.73333 16.4084 10.2583 16.4084 10.5833 16.0834L16.075 10.5917C16.4 10.2667 16.4 9.74172 16.075 9.41672L10.5917 3.91672C10.2667 3.59172 9.74166 3.59172 9.41666 3.91672C9.09166 4.24172 9.09166 4.76672 9.41666 5.09172L13.475 9.16672H4.16666C3.70833 9.16672 3.33333 9.54172 3.33333 10.0001C3.33333 10.4584 3.70833 10.8334 4.16666 10.8334Z" fill="#F0F6FB"/>
    </g>
    <defs>
    <clipPath id="clip0_382_280">
    <rect width="20" height="20" fill="white"/>
    </clipPath>
    </defs>
    </svg>        
    "
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGGroup
                              clipPath="clip0_382_280"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M4.16666 10.8334H13.475L9.40833 14.9001C9.08333 15.2251 9.08333 15.7584 9.40833 16.0834C9.73333 16.4084 10.2583 16.4084 10.5833 16.0834L16.075 10.5917C16.4 10.2667 16.4 9.74172 16.075 9.41672L10.5917 3.91672C10.2667 3.59172 9.74166 3.59172 9.41666 3.91672C9.09166 4.24172 9.09166 4.76672 9.41666 5.09172L13.475 9.16672H4.16666C3.70833 9.16672 3.33333 9.54172 3.33333 10.0001C3.33333 10.4584 3.70833 10.8334 4.16666 10.8334Z"
                                fill={
                                  {
                                    "payload": 4293981947,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                            <RNSVGDefs>
                              <RNSVGClipPath
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                name="clip0_382_280"
                              >
                                <RNSVGRect
                                  fill={
                                    {
                                      "payload": 4294967295,
                                      "type": 0,
                                    }
                                  }
                                  height="20"
                                  propList={
                                    [
                                      "fill",
                                    ]
                                  }
                                  width="20"
                                  x="0"
                                  y="0"
                                />
                              </RNSVGClipPath>
                            </RNSVGDefs>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="EditImageForm"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
